### PR TITLE
KAFKA-1372: separate 400 error from 500 error in RestClient

### PR DESCRIPTION
**Pull request by blcksrx**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java [hunk 1]
```diff
@@ -139,6 +139,9 @@
                         Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
                         "Unexpected status code when handling forwarded request: " + responseCode);
             }
+        } catch(ConnectRestException e) {
```
Comment (kirktrue) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java]: Looks like there are some checkstyle violations in the commit.

It's probably choking on not having a space between the `catch` and open parenthesis.

```suggestion
        } catch (ConnectRestException e) {
```

**Review by kirktrue — CHANGES_REQUESTED** [review]

**Comment by blcksrx** [comment]
@kirktrue

**Review by kirktrue — APPROVED** [review]

**Comment by kirktrue** [comment]
Thanks for the PR! It looks good to me. You need to get a committer to look at this in order to get it merged in. Perhaps a `git blame` can turn up someone who's worked on this that you can ping.

**Comment by blcksrx** [comment]
@mimaison

**Comment by mimaison** [comment]
Thanks for the PR @blcksrx ! I'll try to take a look in the next few days

**Comment by C0urante** [comment]
There's also https://github.com/apache/kafka/pull/11844, which addresses the same issue and takes a slightly different approach that preserves the exact status that the client received from the request to the forwarded worker in the response to the REST API.

**Comment by C0urante** [comment]
BTW, probably want to update the ticket number in the title to 13702.

**Comment by mimaison** [comment]
We merged https://github.com/apache/kafka/pull/12320, so closing this PR

**Pull request closed without merging**